### PR TITLE
Add new hint fox at the East Forest well

### PR DESCRIPTION
**Pull request by ScipioWright**
fimsh

### CODE REVIEW

File: src/Patches/GhostHints.cs [hunk 1]
```diff
@@ -334,6 +367,13 @@ public class HintGhost {
                     if (PaletteEditor.CelShadingEnabled && PaletteEditor.ToonFox != null) {
                         NewGhostFox.transform.GetChild(2).GetChild(1).GetComponent<SkinnedMeshRenderer>().material = PaletteEditor.ToonFox.GetComponent<MeshRenderer>().material;
                     }
+                    
+                    if (HintGhost.Fishing == true) {
```
Comment (silent-destroyer) [inline src/Patches/GhostHints.cs:371]:
```suggestion
                    if (HintGhost.Fishing) {
                        GameObject fishingRod = GameObject.Instantiate(ModelSwaps.FishingRod, HintGhost.FishingRodPos.pos, HintGhost.FishingRodPos.rot);
                        fishingRod.SetActive(true);
```

File: src/Patches/GhostHints.cs [hunk 2]
```diff
@@ -334,6 +367,13 @@ public class HintGhost {
...
+                        GameObject fishingRod = GameObject.Instantiate(ModelSwaps.FishingRod);
+                        fishingRod.SetActive(true);
+                        fishingRod.transform.position = HintGhost.FishingRodPos.pos;
+                        fishingRod.transform.rotation = HintGhost.FishingRodPos.rot;
+                    }
```
Comment (silent-destroyer) [inline src/Patches/GhostHints.cs:376]:
```suggestion
```suggestion
                    if (HintGhost.Fishing) {
                        GameObject fishingRod = GameObject.Instantiate(ModelSwaps.FishingRod, HintGhost.FishingRodPos.pos, HintGhost.FishingRodPos.rot);
                        fishingRod.SetActive(true);
                    }
```

File: src/Patches/GhostHints.cs [hunk 3]
```diff
@@ -61,6 +63,34 @@ public class HintGhost {
                 HintedItem = "";
                 OptionalCheckID = "";
             }
+            public HintGhost(string name, string sceneName, Vector3 position, Quaternion rotation, NPC.NPCAnimState animState, string dialogue, bool fishing, TransformData fishingRodPos) {
```
Comment (silent-destroyer) [inline src/Patches/GhostHints.cs:66]: Can we add these as optional parameters for the existing constructors? i.e.
`bool fishing = false, TransformData fishingRodPos = new TransformData()`

Comment (ScipioWright) [reply src/Patches/GhostHints.cs]: Done

File: src/Patches/GhostHints.cs [hunk 4]
```diff
@@ -269,10 +275,11 @@ public class HintGhost {
...
             },
-            { "East Belltower Lower", new List<HintGhost>() {
-                new HintGhost("Hint Ghost East Belltower Lower", "Forest Belltower", new Vector3(500.9258f, 13.9394f, 63.79896f), new Quaternion(0f, 0.9659258f, 0f, 0.2588191f), NPC.NPCAnimState.SIT, $"#uh lahdur brOk ahnd #Ar R ehnehmEz owtsId, buht #is stahJoo\nawfurz sAftE.") }
+            { "East Forest 1", new List<HintGhost>() {
+                new HintGhost("Hint Ghost East Belltower Lower", "Forest Belltower", new Vector3(500.9258f, 13.9394f, 63.79896f), new Quaternion(0f, 0.9659258f, 0f, 0.2588191f), NPC.NPCAnimState.SIT, $"#uh lahdur brOk ahnd #Ar R ehnehmEz owtsId, buht #is stahJoo\nawfurz sAftE."),
+                new HintGhost("Hint Ghost East Forest 1", "East Forest Redux", new Vector3(75.5f, 4f, 52f), new Quaternion(0f, -1f, 0f, 0f), NPC.NPCAnimState.FISHING, $"plEz dOnt %rO koinz in #uh wehl, yool skAr uhwA\n#uh fi$.", true, new TransformData(new Vector3(75.5f, 5.1f, 51.7f), new Quaternion(0f, 0.7071f, 0f, 0.7071f), new Vector3(1f, 1f, 1f))) }
```
Comment (silent-destroyer) [inline src/Patches/GhostHints.cs:280]:
```suggestion
            { "East Forest Entrance", new List<HintGhost>() {
                new HintGhost("Hint Ghost East Forest Entrance 1", "Forest Belltower", new Vector3(500.9258f, 13.9394f, 63.79896f), new Quaternion(0f, 0.9659258f, 0f, 0.2588191f), NPC.NPCAnimState.SIT, $"#uh lahdur brOk ahnd #Ar R ehnehmEz owtsId, buht #is stahJoo\nawfurz sAftE."),
                new HintGhost("Hint Ghost East Forest Entrance 2", "East Forest Redux", new Vector3(75.5f, 4f, 52f), new Quaternion(0f, -1f, 0f, 0f), NPC.NPCAnimState.FISHING, $"plEz dOnt %rO koinz in #uh wehl, yool skAr uhwA\n#uh fi$.", true, new TransformData(new Vector3(75.5f, 5.1f, 51.7f), new Quaternion(0f, 0.7071f, 0f, 0.7071f), new Vector3(1f, 1f, 1f))) }
```

File: src/Patches/GhostHints.cs [hunk 5]
```diff
@@ -269,10 +275,11 @@ public class HintGhost {
...
             },
-            { "East Belltower Upper", new List<HintGhost>() {
+            { "East Forest 2", new List<HintGhost>() {
                 new HintGhost("Hint Ghost East Belltower Upper 1", "Forest Belltower", new Vector3(500.3264f, 62.012f, 107.5831f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.GAZE, $"di^ daw^! doo yoo hahv wuht it tAks?"),
                 new HintGhost("Hint Ghost East Belltower Upper 2", "Forest Belltower", new Vector3(593.9467f, 14.0052f, 84.43121f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.IDLE, $"wow... yoo did it!"),
                 new HintGhost("Hint Ghost East Belltower Upper 3", "East Forest Redux Laddercave", new Vector3(159.0245f, 17.89421f, 78.52466f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.PRAY, $"#Arz uh baws uhhehd... buht hE hahz por vi&uhn.")}
```
Comment (silent-destroyer) [inline src/Patches/GhostHints.cs:285]:
```suggestion
            { "East Forest Exit", new List<HintGhost>() {
                new HintGhost("Hint Ghost East Forest Exit 1", "Forest Belltower", new Vector3(500.3264f, 62.012f, 107.5831f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.GAZE, $"di^ daw^! doo yoo hahv wuht it tAks?"),
                new HintGhost("Hint Ghost East Forest Exit 2", "Forest Belltower", new Vector3(593.9467f, 14.0052f, 84.43121f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.IDLE, $"wow... yoo did it!"),
                new HintGhost("Hint Ghost East Forest Exit 3", "East Forest Redux Laddercave", new Vector3(159.0245f, 17.89421f, 78.52466f), new Quaternion(0f, 0.9238796f, 0f, -0.3826834f), NPC.NPCAnimState.PRAY, $"#Arz uh baws uhhehd... buht hE hahz por vi&uhn.")}
```

**Review by silent-destroyer — APPROVED** [review]

**Pull request merged**